src/main/java/org/altbeacon/beacon/BeaconManager.java

```diff
@@ -109,9 +109,9 @@
     protected static BeaconManager client = null;
     private final ConcurrentMap<BeaconConsumer, ConsumerInfo> consumers = new ConcurrentHashMap<BeaconConsumer,ConsumerInfo>();
     private Messenger serviceMessenger = null;
-    protected RangeNotifier rangeNotifier = null;
+    protected final List<RangeNotifier> rangeNotifiers = new CopyOnWriteArrayList<>();
```

Comment:
I think it would be good if these collections were based on `Set` not `List`.  Because if folks repeatedly call `addXxxNotifier()` with the same object, it could cause trouble.  Making this change could protect against this.  Perhaps use `ConcurrentHashMap.newKeySet()`?